clang-tools-extra/clangd/tool/ClangdMain.cpp

```diff
@@ -472,8 +470,14 @@ int main(int argc, char *argv[]) {
   CcOptions.EnableAllTypes = AllTypes;
   CcOptions.DisableCCTypeChecker = OptDiableCCTypeChecker;
   std::string Malloc = OptMalloc.getValue();
-  if (!Malloc.empty())
-    boost::split(CcOptions.AllocatorFunctions, Malloc, boost::is_any_of(","));
+  if (!Malloc.empty()) {
+    char *as_chr = strdup(Malloc.c_str());
+    char *tok = strtok(as_chr, ",");
+    while (tok != nullptr) {
+      CcOptions.AllocatorFunctions.push_back(tok);
+      tok = strtok(nullptr, ",");
```

Comment:
I'm a little concerned about using C functions for string manipulation. For example `strdup` calls `malloc` to make a copy of the string. It might be that `c_str` does too. Then we have memory leaks. `strtok` results in sharing the same underlying buffer amongst several strings.

What's the standard C++ of doing string splitting? Don't use Boost -- is there a STL function to do this? Maybe look in the C++ book you have?